packages/backend/src/peripherals/database/StateUpdateRepository.ts

```diff
@@ -0,0 +1,166 @@
+import { PedersenHash } from '@explorer/crypto'
+import { AssetBalance, AssetId, OraclePrice } from '@explorer/encoding'
+import { Knex } from 'knex'
+import {
+  AssetBalanceJson,
+  PositionRow,
+  PriceRow,
+  StateUpdateRow,
+} from 'knex/types/tables'
+
+import { Hash256 } from '../../model'
+import { Logger } from '../../tools/Logger'
+
+export interface StateUpdateRecord {
+  id: number
+  blockNumber: number
+  factHash: Hash256
+  rootHash: PedersenHash
+  timestamp: number
+}
+
+export interface PositionRecord {
+  positionId: number
+  publicKey: string
+  collateralBalance: bigint
+  balances: readonly AssetBalance[]
+}
+
+export class StateUpdateRepository {
+  constructor(private knex: Knex, private logger: Logger) {
+    this.logger = logger.for(this)
+  }
+
+  async add({ stateUpdate, positions, prices }: StateUpdateBundle) {
+    await this.knex.transaction(async (trx) => {
+      await this.knex('state_updates')
```

Comment:
I think that writing `trx('...')` is cleaner than `this.knex('...').transacting(trx)`, but if you feel differently we can keep the longer version.